src/doc/en/developer/packaging.rst

```diff
@@ -1366,3 +1207,220 @@ must meet the following requirements:
   :ref:`chapter-github`.
 
 
+.. _section-spkg-patching:
+
+Modifying third-party code
+==========================
+
+In the Sage distribution, we try to use unpatched original upstream tarballs
+of stable versions of third-party packages whenever possible.
+
+Sometimes, however, modifications are necessary, either to fix a bug or
+to make the package build on the platforms supported by Sage.
+
+
+.. _section-spkg-patch-or-repackage:
+
+When to patch, when to repackage, when to autoconfiscate
+--------------------------------------------------------
+
+- First check whether there is already a newer stable version of the package
+  available that fixes the problem. In this case, try to upgrade the package.
+
+- Check if Debian or another distribution already provides patches
+  for upstream.  Use them if possible, don't reinvent the wheel.
+
+- If the upstream project is maintained on GitHub, check if there is a Pull
+  Request that can be imported; see :ref:`section-spkg-patch-from-pr` below.
+
+- Sometimes it may seem as if you need to patch a (hand-written)
+  ``Makefile`` because it "hard-codes" some paths or compiler flags:
+
+  .. CODE-BLOCK:: diff
+
+      --- a/Makefile
+      +++ b/Makefile
+      @@ -77,7 +77,7 @@
+       # This is a Makefile.
+       # Handwritten.
+
+      -DESTDIR = /usr/local
+      +DESTDIR = $(SAGE_ROOT)/local
+       BINDIR   = $(DESTDIR)/bin
+       INCDIR   = $(DESTDIR)/include
+       LIBDIR   = $(DESTDIR)/lib
+
+  Don't use patching for that.  Makefile variables can be overridden
+  from the command-line.  Just use the following in ``spkg-install``:
+
+  .. CODE-BLOCK:: bash
+
+      $(MAKE) DESTDIR="$SAGE_ROOT/local"
+
+- If the upstream Makefile does not build shared libraries,
+  don't bother trying to patch it.
+
+  Autoconfiscate the package instead and use the standard facilities
+  of Automake and Libtool.  This ensures that the shared library build
+  is portable between Linux and macOS.
+
+- If you have to make changes to ``configure.ac`` or other source
+  files of the autotools build system (or if you are autoconfiscating
+  the package), then you can't use patching; make a :ref:`modified
+  tarball <section-spkg-src>` instead.
+
+- If the patch would be huge, don't use patching.  Make a
+  :ref:`modified tarball <section-spkg-src>` instead.
+
+- Otherwise, :ref:`maintain a set of patches
+  <section-spkg-patch-maintenance>`.
+
+
+.. _section-spkg-patch-from-pr:
+
+Patching sources by importing a pull request from GitHub
+--------------------------------------------------------
+
```

Comment:
Instead of 

"Patching sources by importing a pull request from GitHub"

and

"Manual preparation of patches"

How about

"Preparing a patch by importing a pull request from GitHub"

and

"Preparing  a patch manually"

so that they look parallel?

Reply:
Good suggestion, done.